plaso/parsers/fish_history.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+"""Parser for fish history files."""
+import io
+import re
+
+import yaml
+
+from dfdatetime import posix_time as dfdatetime_posix_time
+
+from plaso.containers import events
+from plaso.containers import time_events
+from plaso.lib import errors
+from plaso.lib import definitions
+from plaso.parsers import logger
+from plaso.parsers import manager
+from plaso.parsers import interface
+
+
+class FishHistoryEventData(events.EventData):
+  """Fish history log event data.
+
+  Attributes:
+    command (str): command that was executed.
+  """
+
+  DATA_TYPE = 'fish:history:command'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(FishHistoryEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.command = None
+
+
+class FishHistoryParser(interface.FileObjectParser):
+  """Parses events from Fish history files."""
+
+  NAME = 'fish_history'
+  DATA_FORMAT = 'Fish history file'
+
+  _ENCODING = 'utf-8'
+
+  _FILENAME = 'fish_history'
+  _YAML_FORMAT_RE_1 = re.compile(r'^- cmd: \S+')
+  _YAML_FORMAT_RE_2 = re.compile(r'  when: [0-9]{9}')
+
+  def ParseFileObject(self, parser_mediator, file_object):
+    """Parses a Fish history file from a file-like object
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      file_object (dfvfs.FileIO): a file-like object.
+
+    Raises:
+      ParseError: if the file is not valid YAML or is not a fish history entry
+      UnableToParseFile: when the file cannot be parsed.
+      ValueError: if the file object is missing
+    """
+    if not file_object:
+      raise ValueError('Missing file object.')
+
+    filename = parser_mediator.GetFilename()
+
+    if filename != self._FILENAME:
+      raise errors.UnableToParseFile('Not a fish history file.')
+
+    file_contents = file_object.read()
```

Comment:
This will exhaust memory if the file content size is large, maybe there should be a check first of this size?

```
file_size = file_object.get_size()
if file_size > self._MAXIMUM_FILE_SIZE:
  raise errors.UnableToParseFile('Not a fish history file.')
```

Reply:
Added a file size limit of 50MB which should be sufficient for a file of this type.   An extraction warning will be raised if the file size is larger than this.